src/nvim/ex_docmd.c

```diff
@@ -76,6 +76,16 @@
 #include "nvim/event/rstream.h"
 #include "nvim/event/wstream.h"
 
+/// The scope of a command.
+///
+/// The higher a number, the deeper a scope.
+enum {
+  SCOPE_EDITOR,     ///< Affects the entire instance of NeoVim.
+  SCOPE_TABPAGE,    ///< Affects on tab page.
+  SCOPE_WINDOW,     ///< Affects one window.
+  NUMBER_OF_SCOPES  ///< Total number of scoped defined.
```

Comment:
Enumerations start at 0 and every consecutive item is one larger, so the last item will be the number of items. This way is automatic, and if someone were to introduce a new scope the number of scopes would adjust automatically. Unlikely to ever happen, but I still prefer in general the compiler to take care of constants instead of me.